crates/codegen/language/definition/src/model/utils/mod.rs

```diff
@@ -1,5 +1,78 @@
+use crate::model::{self, Item};
+use semver::Version;
+use std::collections::BTreeSet;
+
 mod identifier;
 mod version_specifier;
 
 pub use identifier::*;
 pub use version_specifier::*;
+
+/// Collects all versions that changes the language grammar in a breaking way.
+pub fn collect_breaking_versions(lang: &model::Language) -> BTreeSet<Version> {
```

Comment:
Similarly, I wonder if this should also be a `Language::breaking_versions()` extension?

Reply:
I like idea of the `collect_breaking_versions` as a utility function that accepts a heap of (just) data, does some processing on it and returns the result; otherwise `Language::breaking_versions()` feels like a cheap getter/accessor, which is not the case here.

I don't have strong feelings about it; if we want to use an associated function, I'd rather we not skip `collect` in the name to imply more work/processing.

Do you want me to add this as a `Language` associated function?